### DISCUSSION TITLE
Import from Commento

### DISCUSSION THREAD

**Discussion by hiensarahly** [walinejs/waline#174] [Q&A] [answered]
Can I import comments from Commento into Waline?

**Comment by lizheming** [comment, marked as answer]
Waline has not support import from commento data yet. You can send me a commento export example, then I can write commento importer base on it.

**Reply by hiensarahly** [reply]
Here is the JSON data export
comment:

```json
{
            "commentHex": "97cfc6869eefd9d1f90a32563ad5dd8acab69adeeccb4b62b86158b10fe9b627",
            "domain": "example.com",
            "url": "/sample-url/",
            "commenterHex": "62bff32833af4e7ea541fb8eedeb59d6d0731ed56482486730b4af1b787f3a85",
            "markdown": "Sample comment",
            "html": "",
            "parentHex": "root",
            "score": 0,
            "state": "approved",
            "creationDate": "2020-07-13T00:04:07.650839Z",
            "direction": 0
        }
```

commenter:

```json
{
            "commenterHex": "42230a1996821c31ad5bb104baf3bf8afe9b190d9dae1769aa7eeabe5f17baa5",
            "email": "example@email.com",
            "name": "Name",
            "link": "undefined",
            "photo": "undefined",
            "provider": "commento",
            "joinDate": "2020-07-02T04:35:17.104629Z",
            "isModerator": false
        }
```

**Reply by lizheming** [reply]
Thank you very much, I'll add commento import support asap!

**Reply by hiensarahly** [reply]
thank u very much,😋 i’m happy to know!

**Reply by lizheming** [reply]
Hello @hiensarahly  commento import support now. You can go to https://waline.js.org/migration/tool.html and handle as following try it now.

1. Go to documentation and select `Commento` in the first select input.
2. Select which waline storage you want to import in the second select input.
3. Paste JSON data which you export from commento
4. Click the green button at the bottom of textarea input.
5. You'll download a file after data transform success. Then You can go to waline storage import data by using the file.

![image](https://user-images.githubusercontent.com/424491/111897349-6c15f480-8a5a-11eb-97d7-1510e45966f0.png)

**Reply by hiensarahly** [reply]
Hello thanks so much @lizheming it works great, I’ll move over!
